src/main/common/sensor_alignment.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "axis.h"
+#include "maths.h"
+
+typedef enum {
+    ALIGN_DEFAULT = 0, // driver-provided alignment
+
+    CW0_DEG = 1,
+    CW90_DEG = 2,
+    CW180_DEG = 3,
+    CW270_DEG = 4,
+    CW0_DEG_FLIP = 5,
+    CW90_DEG_FLIP = 6,
+    CW180_DEG_FLIP = 7,
+    CW270_DEG_FLIP = 8,
+
+    ALIGN_CUSTOM = 9, // arbitrary sensor angles, e.g. for external sensors
+} sensor_align_e;
+
+typedef union sensorAlignment_u {
+    // value order is the same as axis_e
+
+    int16_t raw[XYZ_AXIS_COUNT];
+    struct {
+        int16_t roll;
+        int16_t pitch;
+        int16_t yaw;
+    } values;
+} sensorAlignment_t;
+
+
+#define CUSTOM_ALIGN_CW0_DEG         ((sensorAlignment_t){ .values = { .roll = 0, .pitch = 0,   .yaw = 0    }})
```

Comment:
May be better to use buildAlignmentFromLegacyRotation - the definition will be only in one place. Or combine it with bitpacking to use CUSTOM_ to build array of possible enumerated orientations

Reply:
`buildAlignmentFromLegacyRotation` is now `buildAlignmentFromStandardAlignment` and only used in test code and may be deleted or moved back as needed.  Only one of these values is ever used at firmware compile time anyway so it's probably overkill to do as you suggest.

Reply:
done, see "Ensure gyro/mag custom alignment settings follow the enum variations" commit - https://github.com/betaflight/betaflight/pull/8474/commits/36bf663169faf528254bb7e12d375ab4be95cbb0